### PR TITLE
ci: run stable tests partially

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

stabe tests take about 30 min. it seems too long. 

#### Summary of Changes

use nextest to run them partially. the steps looks like
Screenshot 2023-04-12 at 11 50 10 AM

I separated packages into two group, A) use nextest and B) not use nextest
you may see lots of jobs in the partitions. the main idea is that if we have N steps. the last must run group B with normal cargo test command. the other steps will run group A and separate them to 1/N-1, 2/N-1, 3/N-1 ... N-1/N-1.

### CODE REVIEW

**Comment by yihau** [comment]
the latest build. if we have sufficient agents, it will only take 8 mins to build 😈 
Screenshot 2023-04-13 at 3 33 08 PM

**Review by t-nelson — CHANGES_REQUESTED** [review]
can you factor out the refactor/cleanup changes to a separate PR so the partitioning changes are better isolated?

**Comment by yihau** [comment]
sure! I extracted some cleanup logic to PR 31196. I will rebase this PR when the PR 31196 merge.

**Comment by t-nelson** [comment]
is this ready for another review pass?

**Comment by yihau** [comment]
Thank you for asking! yeah, I forgot to ping you guys again. this PR is ready for another look 👀

File: .buildkite/scripts/build-stable.test.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -e
+here=$(dirname "$0")
+
+# shellcheck source=.buildkite/scripts/func-assert-eq.sh
+source "$here"/func-assert-eq.sh
+
+want=$(cat <<EOF
```
[the remark this answers is not in the document]
Comment (yihau) [reply .buildkite/scripts/build-stable.test.sh]: yeah! it looks so good! fixed [105db43](https://github.com/solana-labs/solana/pull/31151/commits/105db43f394ff1d76a471027f05ea99e992cc8af)

File: .buildkite/scripts/common.sh [hunk 2]
```diff
@@ -35,11 +36,37 @@ step() {
...
+    maybe_retry=$(
+      cat <<EOF | indent 2
+retry:
+  automatic:
+    - limit: $retry
+EOF
+    )
+  fi
+
+  cat <<EOF | indent | sed '/EMPTY_LINE/d'
```
[the remark this answers is not in the document]
Comment (yihau) [reply .buildkite/scripts/common.sh]: oh. the EMPTY_LINE is not the real EMPTY_LINE. it is a denote for me to remove some unset attributes. I try to rename it here [453b9c3](https://github.com/solana-labs/solana/pull/31151/commits/453b9c33c7362a0f59bd6368a19d2cf02413a3e9)

File: ci/stable/run-partition.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+source "$here"/../common/shared-functions.sh
+
+#shellcheck source=ci/common/limit-threads.sh
+source "$here"/../common/limit-threads.sh
+
+#shellcheck source=ci/stable/common.sh
+source "$here"/common.sh
+
+# check partition info
+N=${1:-"$BUILDKITE_PARALLEL_JOB"} # BUILDKITE_PARALLEL_JOB from 0 to (BUILDKITE_PARALLEL_JOB_COUNT - 1)
```
[the remark this answers is not in the document]
Comment (yihau) [reply ci/stable/run-partition.sh]: sure! 
rename N => INDEX [be33cc9](https://github.com/solana-labs/solana/pull/31151/commits/be33cc9276466e6f6a047ad8d9a637ca6758afe9)
rename M => LIMIT [bc38a00](https://github.com/solana-labs/solana/pull/31151/commits/bc38a0047a3786e48d8255f41571bebd9a728046)

File: ci/stable/run-partition.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+if [ -z "$N" ]; then
+  echo "N is not set"
+  exit 1
+fi
```
[the remark this answers is not in the document]
Comment (yihau) [reply ci/stable/run-partition.sh]: you're the king 👑 this way is so awesome! [ef9ae51](https://github.com/solana-labs/solana/pull/31151/commits/ef9ae51848a8b924d69c37d8dd345ba99af3f90c)

File: .buildkite/scripts/common.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (t-nelson) [reply .buildkite/scripts/common.sh]: oh i thought it was to work around yaml's stupid assumptions about empty lines 😅

**Review by t-nelson — APPROVED** [review]
:shipit:

**Pull request merged**